kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala

```diff
@@ -1497,6 +1497,47 @@ class ScanSuite extends AnyFunSuite with TestUtils with ExpressionTestUtils with
         .getScanBuilder(tableClient).withFilter(tableClient, nonEligibleFilter).build()
         .getScanFiles(tableClient))
   }
+
+  test("data skipping - prune schema correctly for various predicates") {
+    def structTypeToLeafColumns(
+        schema: StructType, parentPath: Seq[String] = Seq()): Set[Column] = {
+      schema.fields().asScala.flatMap { field =>
+        field.getDataType() match {
+          case nestedSchema: StructType =>
+            assert(nestedSchema.fields().size() > 0,
+              "Schema should not have field of type StructType with no child fields")
+            structTypeToLeafColumns(nestedSchema, parentPath ++ Seq(field.getName()))
+          case _ =>
+            Seq(new Column(parentPath.toArray :+ field.getName()))
+        }
+      }.toSet
+    }
+    def verifySchema(expectedReadCols: Set[Column]): StructType => Unit = { readSchema =>
+      assert(structTypeToLeafColumns(readSchema) == expectedReadCols)
+    }
+    val path = goldenTablePath("data-skipping-basic-stats-all-types")
+    // Map of expression -> expected read columns
+    Map(
+      equals(col("as_int"), ofInt(0)) ->
+        Set(nestedCol("minValues.as_int"), nestedCol("maxValues.as_int")),
+      lessThan(col("as_int"), ofInt(0)) -> Set(nestedCol("minValues.as_int")),
+      greaterThan(col("as_int"), ofInt(0)) -> Set(nestedCol("maxValues.as_int")),
+      greaterThanOrEqual(col("as_int"), ofInt(0)) -> Set(nestedCol("maxValues.as_int")),
+      lessThanOrEqual(col("as_int"), ofInt(0)) -> Set(nestedCol("minValues.as_int")),
+      new And(
+        lessThan(col("as_int"), ofInt(0)),
+        greaterThan(col("as_long"), ofInt(0))
+      ) -> Set(nestedCol("minValues.as_int"), nestedCol("maxValues.as_long"))
+    ).foreach { case (predicate, expectedCols) =>
+      val tableClient = tableClientVerifyJsonParseSchema(verifySchema(expectedCols))
+      collectScanFileRows(
```

Comment:
Are you doing anything with the result of this read?

Reply:
No just checking that the overridden tableClient doesn't throw an error. and collectScanFileRows is just so the iterators --> seq